shopBuddy/shopBuddy/ItemDetailViewController.m

```diff
@@ -9,25 +9,36 @@
 #import "Item.h"
 #import "UIImageView+AFNetworking.h"
 #import "APIManager.h"
+#import "ShoppingList.h"
+#import "ShoppingList+Persistent.h"
 
 @interface ItemDetailViewController ()
-@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
-@property (weak, nonatomic) IBOutlet UILabel *brandLabel;
-@property (weak, nonatomic) IBOutlet UIImageView *itemImage;
-@property (weak, nonatomic) IBOutlet UITextView *descriptionView;
+{
+    NSArray<ShoppingList*> *lists;
+    __weak IBOutlet UITextView *descriptionView;
+    __weak IBOutlet UILabel *titleLabel;
+    __weak IBOutlet UILabel *brandLabel;
+    __weak IBOutlet UIImageView *itemImage;
+    __weak IBOutlet UIButton *addItemToListButton;
+    
+}
 
 @end
 
 @implementation ItemDetailViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    self.descriptionView.scrollEnabled=YES;
+    descriptionView.scrollEnabled=YES;
     if (self.item != nil) {
         [self _populateView];
     } else {
         [self _callAPI];
     }
+    [ShoppingList fetchListsByUser:[PFUser currentUser] withCompletion:^(NSArray *lists, NSError *error) {
+        self->lists = lists;
+        [self _makeMenu];
+    }];
```

Comment:
This is still unresolved. `self` should not be used inside async blocks to avoid retention cycles.
You should do something like:
```
__weak __typeof__(self) weakSelf = self;
[ShoppingList fetchListsByUser:[PFUser currentUser] withCompletion:^(NSArray *lists, NSError *error) {
        [weakSelf _loadShoppingLists:lists error:error];
}

- (void)_loadShoppingLists:(NSArray *)lists error:(NSError *)error {
        ...
}
```